lib/diagnostic.rb

```diff
@@ -6,37 +6,36 @@ class Diagnostic
   # Write ruby code to return the longest entry in the list supplied.
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
-  def longest(list)
-    # Your code goes here
-  end
+
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.reduce(:+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
+
   def some_odd(numbers)
-    # Your code goes here
+    numbers.any?{|number| number.odd?}
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # list of numbers are even
   def every_even(numbers)
-    # Your code goes here
+    number.all? {|number| number.odd?}
```

Comment:
I think this one works unless it gets the number 0 passed to it for some reason. Here's another way we can do it:

`numbers.all?(&:even?)`